components/layout/construct.rs

```diff
@@ -1161,11 +1064,28 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
 
     /// Builds a flow for a node with `display: table`. This yields a `TableWrapperFlow` with
     /// possibly other `TableCaptionFlow`s or `TableFlow`s underneath it.
-    fn build_flow_for_table_wrapper(&mut self, node: &ConcreteThreadSafeLayoutNode, float_value: float::T)
-                                    -> ConstructionResult {
-        let fragment = Fragment::new(node, SpecificFragmentInfo::TableWrapper, self.layout_context);
-        let mut wrapper_flow: FlowRef = Arc::new(
-            TableWrapperFlow::from_fragment(fragment, FloatKind::from_property(float_value)));
+    fn build_flow_for_table(&mut self, node: &ConcreteThreadSafeLayoutNode, float_value: float::T)
+                            -> ConstructionResult {
+        let mut legalizer = Legalizer::new();
+
+        let table_style = node.style(self.style_context());
+        let wrapper_style = self.style_context()
+                                .stylist
+                                .precomputed_values_for_pseudo(&PseudoElement::ServoTableWrapper,
```

Comment:
This pattern is becoming increasingly common, maybe add a helper for it in `SharedStyleContext`? Can be left as a followup, but please fill it if you decide to do so.